### PR TITLE
hydra-api: flesh out Jobset, JobsetInput schemas; implement DELETE /jobset/{project-id}/{jobset-id}

### PR DESCRIPTION
**Pull request by cole-h**
```
# flake
$ curl -X 'GET' 'http://0:63333/jobset/test/test11' -H 'accept: application/json' | jq
{
  "type": 1,
  "description": "test2",
  "triggertime": null,
  "keepnr": 0,
  "hidden": false,
  "errormsg": "",
  "nixexprinput": "",
  "lastcheckedtime": null,
  "errortime": null,
  "jobsetinputs": {},
  "flake": "github:nixos/nix",
  "enabled": 2,
  "checkinterval": 0,
  "fetcherrormsg": "",
  "project": "test",
  "name": "test11",
  "emailoverride": "",
  "starttime": null,
  "enableemail": false,
  "schedulingshares": 100,
  "nixexprpath": ""
}

# inputs
$ curl -X 'GET' 'http://0:63333/jobset/test/asdf' -H 'accept: application/json' | jq
{
  "fetcherrormsg": "",
  "project": "test",
  "name": "asdf",
  "enabled": 1,
  "checkinterval": 0,
  "enableemail": false,
  "schedulingshares": 1,
  "nixexprpath": "fdsa",
  "emailoverride": "",
  "starttime": null,
  "hidden": false,
  "type": 0,
  "description": "",
  "keepnr": 0,
  "triggertime": null,
  "errortime": null,
  "jobsetinputs": {
    "asdf": {
      "type": "bitbucketpulls",
      "name": "asdf",
      "emailresponsible": false,
      "jobsetinputalts": [
        "asdf"
      ]
    }
  },
  "flake": "",
  "errormsg": "",
  "nixexprinput": "asdf",
  "lastcheckedtime": null
}
```

### CODE REVIEW

File: src/lib/Hydra/Schema/Jobsets.pm [hunk 1]
```diff
@@ -412,12 +412,30 @@ __PACKAGE__->add_column(
 
 my %hint = (
     columns => [
+        "errortime",
+        "lastcheckedtime",
+        "triggertime",
         "enabled",
```
Comment (grahamc) [inline src/lib/Hydra/Schema/Jobsets.pm:418]: Should enabled be a boolean column?

Comment (cole-h) [reply src/lib/Hydra/Schema/Jobsets.pm:418]: No, because it has 4 states: enabled, one-shot, one-at-a-time, and disabled.

Comment (grahamc) [reply src/lib/Hydra/Schema/Jobsets.pm:418]: :upside_down_face: cool

**Comment by grahamc** [comment]
This is some mild API breakage, but probably nothing too bad. Can you write a test for the create / read case? Ignore the dynamic fields like timestamps. I can help with this if you need it.

File: src/lib/Hydra/Controller/User.pm [hunk 2]
```diff
@@ -37,7 +37,11 @@ sub login_POST {
         accessDenied($c, "Bad username or password.")
     }
 
-    currentUser_GET($self, $c);
+    $self->status_found(
+        $c,
+        location => $c->uri_for("current-user"),
+        entity => {}
+    );
```
Comment (grahamc) [inline src/lib/Hydra/Controller/User.pm:44]: Note this is a minor change in the API: clients will need to follow the redirect to get the same response it used to. I'm inclined to think this is a non-issue, but we could replicate the old response body here if needed, by a subsequent PR.

**Review by grahamc — APPROVED** [review]
lgtm, squash and I'll merge?

**Pull request merged**